plugins/aver-hid/fu-aver-hid-device.c

```diff
@@ -306,21 +363,35 @@ fu_aver_hid_device_write_firmware(FuDevice *device,
 	fu_progress_step_done(progress);
 
 	/* ISP_FILE_END */
-	if (!fu_aver_hid_device_isp_file_end(self, error))
+	if (!fu_aver_hid_device_isp_file_end(self, &fw_size, aver_fw_name, error))
 		return FALSE;
 	fu_progress_step_done(progress);
 
-	/* poll for the actual write progress */
-	if (!fu_device_retry_full(device,
-				  fu_aver_hid_device_wait_for_reboot_cb,
-				  FU_AVER_HID_DEVICE_ISP_TIMEOUT,
-				  FU_AVER_HID_DEVICE_GET_STATUS_POLL_INTERVAL,
-				  fu_progress_get_child(progress),
-				  error))
+	/* ISP_START */
+	if (!fu_aver_hid_device_isp_start(self, error))
 		return FALSE;
 	fu_progress_step_done(progress);
+
+	for (guint i = 0; i < FU_AVER_HID_DEVICE_ISP_TIMEOUT; i++) {
+		if (fu_aver_hid_device_isp_status(self, error)) {
```

Comment:
if you're not consuming the error, use a `g_autoptr(GError) error_local = NULL` instead and g_debug() the error_local->message